packages/app/src/cli/commands/app/function/run.ts

```diff
@@ -42,10 +44,43 @@ export default class FunctionRun extends Command {
       path: flags.path,
       userProvidedConfigName: flags.config,
       callback: async (_app, ourFunction) => {
+        let functionExport = DEFAULT_FUNCTION_EXPORT
+
+        if (flags.export !== undefined) {
+          debug(`Using export ${flags.export} from the --export flag.`)
+          functionExport = flags.export
+        } else if (
+          ourFunction.configuration.targeting !== undefined &&
+          ourFunction.configuration.targeting.length > 0
+        ) {
+          const targeting = ourFunction.configuration.targeting
+
+          if (targeting.length > 1 && isTTY()) {
+            let targets = targeting.map((target) => ({
```

Comment:
I'm not 100% on where these come from, but if they're not ordered at the source, it might be worth sorting them first.